CMakeLists.txt

```diff
@@ -108,6 +108,12 @@ dgl_feature_option(
     "dev"
 )
 
+dgl_feature_option(
+    USE_ASAN
+    "Use Address Sanitizer(ASAN) to check memory error"
+    "none"
```

Comment:
let's enable it in `dev`?

Reply:
<img width="816" alt="f1" src="https://github.com/dmlc/dgl/assets/9260628/edd8a0f8-6cf0-42af-b177-292ad8b91c2b">